ubi_manifest/worker/tasks/depsolver/utils.py

```diff
@@ -91,40 +96,36 @@ def get_n_latest_from_content(content, blacklist, modular_rpms=None):
 
 
 def parse_bool_deps(bool_dependency):
-    """Parses bool/rich dependency clause and returns set of names of packages"""
-    # remove all paranthesis from clause
-    _dep = re.sub(r"\(|\)", "", bool_dependency)
+    """Parses boolean/rich dependency clause and returns set of names of packages"""
+
+    # remove enclosing parenthesis
+    _dep = re.sub(OPEN_END_ONE_PAR_REGEX, "", bool_dependency)
```

Comment:
I don't get if this part is actually useful.  In fact I deleted this line and just changed the code to:

```python
to_parse = bool_dependency.split()
```

...and all the tests still pass.

Logically this bit of the code doesn't seem to make sense, as it can strip one level of '()', but who says there is only one?  If (foo) is a valid expression then I guess ((foo)) and (((foo))) are also, so it wouldn't be right to remove just one level of them.

It seems like what makes it work correctly (makes the tests pass) is the "remove all starting and ending paranthesis" bit later, but if that works correctly then I guess there's not really a point in this removing a *single* starting/ending parenthesis here.

Reply:
I just wonder, if a bool dep. can be in form with space near parenthesis:
`( dep1 and dep2 )`
or somehow more messed.
In that case one of item we iterate over will be `(`, and will be changed to empty string by line:
`item = re.sub(OPEN_END_ONE_OR_MORE_PAR_REGEX, "", item)`
and then added to the output set.
Not sure if this is a real concern but I guess it won't harm to be a bit more defensive.